src/utils.ts

```diff
@@ -98,7 +98,8 @@ export function digestAuthHeader(method: string, uri: string, wwwAuthenticate: s
   }
 
   let qop = opts.qop || '';
-  const [ user, pass ] = userpass.split(':');
+  const user = userpass.split(':', 1);
+  const pass = userpass.slice(user[0].length+1);
```

Comment:
```ts
const index = userpass.indexOf(':');
const user = userpass.substring(0, index);
const pass = userpass.substring(index + 1);
```

Find the `:` position index then split out the `user` and `pass` should be better.

Reply:
Edited the pull request to reflect proposed change.